ocs_ci/ocs/fiojob.py

```diff
@@ -148,26 +148,34 @@ def wait_for_job_completion(namespace, timeout, error_msg):
         ocp_pod.wait_for_resource(
             resource_count=1,
             condition=constants.STATUS_COMPLETED,
+            error_condition=constants.STATUS_ERROR,
             timeout=timeout,
             sleep=30
         )
-    except TimeoutExpiredError as ex:
-        # report some high level error as well
-        logger.error(error_msg)
-        ex.message = error_msg
+    except Exception as ex:
```

Comment:
Please catch the exceptions you are excepting to see to let other exceptions to catch by the system and break the test. Exceptions you are raising in wait_for_resource method:
ResourceInUnexpectedState, ValueError and the TimoueOutExpired as before this change

Reply:
The code here catches the exceptions to perform additional logging (fetching pod logs) and then it always raises it again, so that no information about the error is lost. In case of `TimeoutExpiredError` it also updates the error message so that the test failure would be faster to evaluate.

I originally added `ResourceInUnexpectedState` in the list of exceptions as you suggest, as this is a general principle, but then I realized that in this case, I would rather have the additional evidence for every error which could happen.